### PR TITLE
[pipes] databricks termination forwarding

### PR DESCRIPTION
**Pull request by alangenfeld**
Changes the default behavior to forward termination/cancelation to the Databricks run. Adds a flag on the client to toggle this behavior (based on feedback from conversing with user).

While I was adding the toggle it made sense to me to move the hard coded poll interval to client instance option as well. 

addresses https://github.com/dagster-io/dagster/issues/18570 for databricks client 

## How I Tested These Changes

run `pytest python_modules/libraries/dagster-databricks/dagster_databricks_tests/test_pipes.py::test_pipes_client`
ctrl+c
observe the databricks job run was cancelled

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* `master`
  * **PR #18731** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/18731?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/18731?utm_source=stack-comment).

File: python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py [hunk 1]
```diff
@@ -146,27 +152,59 @@ def run(
...
+                self._poll_til_success(context, run_id)
+            except DagsterExecutionInterruptedError:
+                if self.forward_termination:
+                    context.log.info("[pipes] execution interrupted, canceling Databricks job.")
+                    self.client.jobs.cancel_run(run_id)
+                    self._poll_til_terminating(run_id)
+
+        return PipesClientCompletedInvocation(pipes_session)
+
+    def _poll_til_success(self, context: OpExecutionContext, run_id: str):
```
Comment (smackesey) [inline python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py:165]: [1] add -> None annotation

File: python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py [hunk 2]
```diff
@@ -146,27 +152,59 @@ def run(
...
-                time.sleep(_RUN_POLL_INTERVAL)
-        return PipesClientCompletedInvocation(pipes_session)
+            elif run.state.life_cycle_state == jobs.RunLifeCycleState.INTERNAL_ERROR:
+                raise DagsterPipesExecutionError(
+                    f"Error running Databricks job: {run.state.state_message}"
+                )
+
+            time.sleep(self.poll_interval_seconds)
+
+    def _poll_til_terminating(self, run_id: str):
```
Comment (smackesey) [inline python_modules/libraries/dagster-databricks/dagster_databricks/pipes.py:194]: [1]

**Review by smackesey — APPROVED** [review]
cool

**Pull request merged**